Parallel builds stopped working

### Before reporting an issue

- [X] I have read and understood the above terms for submitting issues, and I understand that my issue may be closed without action if I do not follow them.

### Area

admin/ui

### Describe the bug

Using parallel build to compile Keycloak has stopped working between 21.0.0 and 22.0.0 releases.


### Version

main

### Expected behavior

Compilation should succeed.

### Actual behavior

Compiling fails with errors


```
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-js-parent: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-js-admin-client: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-account-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
[ERROR] Failed to execute goal com.github.eirslett:frontend-maven-plugin:1.14.2:pnpm (pnpm-install) on project keycloak-admin-ui: Failed to run task: 'pnpm install --prefer-offline --frozen-lockfile --ignore-scripts' failed. org.apache.commons.exec.ExecuteException: Process exited with an error: 1 (Exit value: 1) -> [Help 1]
```

### How to Reproduce?

Compile Keycloak by running

```
mvn -T4C clean install -DskipTestsuite -DskipExamples -DskipTests
```

### Anything else?

I've used `git bisect` to track the commit which makes parallel builds fail and it seems to be #17401 and commit  https://github.com/keycloak/keycloak/commit/6d2e57f93a297360c4ebb6082dcd7fc87ccf2448.

In parallel build `pnpm install` also gets executed several times in parallel

```
/home/tsaarni/work/keycloak/node/node_modules/pnpm/bin/pnpm.cjs install --prefer-offline --frozen-lockfile --ignore-scripts
```

and it fails 

```
ERR_PNPM_LINKING_FAILED  Error: ENOTEMPTY: directory not empty, rmdir '/home/tsaarni/work/keycloak/js/node_modules/.pnpm/file+apps+admin-ui_@types+react@18.2.34_react-monaco-editor@0.51.0/node_modules/admin-ui
```

As I mentioned in #24537 that I suspect the same problem might be happening when opening Keycloak in vscode with [Language Support for Java(TM) by Red Hat](https://marketplace.visualstudio.com/items?itemName=redhat.java).

Here is how to reproduce:

1. I've built Keycloak fully with maven on command line NOT using parallel build (=required for build to succeed)
2. I open the project in vscode and wait for project to compile
3. I try to launch the project either (a) within vscode using [`IDELauncher`](https://github.com/keycloak/keycloak/blob/main/quarkus/README.md#running-from-your-ide) class, or by running (b) [quarkus:dev mode](https://github.com/keycloak/keycloak/blob/main/quarkus/README.md#contributing)

Here is the output from step (2) taken from Java language server logs https://gist.github.com/tsaarni/a021e6bba858b3506c9b9bccf955d2c3. It includes several errors while running `pnpm-install` goal.

I suspect this causes build to be in non-consistent state, since it was unfinished. Running keycloak fails with various errors caused by this, for example:

```
2023-11-08 16:47:42,745 ERROR [org.keycloak.services.error.KeycloakErrorHandler] (executor-thread-3) Uncaught server error: org.keycloak.theme.FreeMarkerException: Failed to process template index.ftl
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:52)
        at org.keycloak.services.resources.admin.AdminConsole.getMainPage(AdminConsole.java:352)
        at org.keycloak.services.resources.admin.AdminConsole$quarkusrestinvoker$getMainPage_d508fd32a638b899af4d52f90082abf4f245fdb3.invoke(Unknown Source)
        at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)
        at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)
        at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:145)
        at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)
        at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)
        at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)
        at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)
        at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)
        at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)
        at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: freemarker.template.TemplateNotFoundException: Template not found for name "index.ftl".
The name was interpreted by this TemplateLoader: org.keycloak.theme.freemarker.DefaultFreeMarkerProvider$ThemeTemplateLoader@774f875a.
        at freemarker.template.Configuration.getTemplate(Configuration.java:2957)
        at freemarker.template.Configuration.getTemplate(Configuration.java:2777)
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.getTemplate(DefaultFreeMarkerProvider.java:66)
        at org.keycloak.theme.freemarker.DefaultFreeMarkerProvider.processTemplate(DefaultFreeMarkerProvider.java:45)
        ... 12 more
```

In order to fix the build, i must **close** vscode (to avoid language server build), make a full clean build with maven on command line and then run via maven `quarkus:dev`mode.  When I open vscode again to work with the code, the problem will repeat. Therefore also running `IDELaucher` and vscode debugger is out of question

--- a/themes/pom.xml
+++ b/themes/pom.xml
@@ -22,6 +22,19 @@
         <maven.build.cache.exclude.4>${project.basedir}/src/main/resources/theme/keycloak.v2/account/src/web_modules</maven.build.cache.exclude.4>
     </properties>
 
+    <dependencies>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-admin-ui</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.keycloak</groupId>
+            <artifactId>keycloak-account-ui</artifactId>
+            <version>${project.version}</version>
+        </dependency>
+    </dependencies>
+
     <build>
         <resources>
             <resource>

--- a/themes/pom.xml
+++ b/themes/pom.xml
@@ -26,12 +26,10 @@
         <dependency>
             <groupId>org.keycloak</groupId>
             <artifactId>keycloak-admin-ui</artifactId>
-            <version>${project.version}</version>
         </dependency>
         <dependency>
             <groupId>org.keycloak</groupId>
             <artifactId>keycloak-account-ui</artifactId>
-            <version>${project.version}</version>
         </dependency>
     </dependencies>
 

